pkg/scheduler/ugm/group_tracker_test.go

```diff
@@ -172,32 +174,49 @@ func TestGTSetMaxLimits(t *testing.T) {
 	// Initialize ugm
 	GetUserManager()
 	user := security.UserGroup{User: "test", Groups: []string{"test"}}
-	groupTracker := newGroupTracker(user.User)
+	groupTracker := newGroupTracker(user.User, newUGMEvents(mock.NewEventSystemDisabled()))
 	usage1, err := resources.NewResourceFromConf(map[string]string{"mem": "10M", "vcore": "10"})
 	if err != nil {
 		t.Errorf("new resource create returned error or wrong resource: error %t, res %v", err, usage1)
 	}
 
-	result := groupTracker.increaseTrackedResource(hierarchy1, TestApp1, usage1, user.User)
+	result := groupTracker.increaseTrackedResource(path1, TestApp1, usage1, user.User)
 	if !result {
-		t.Fatalf("unable to increase tracked resource: queuepath %+q, app %s, res %v", hierarchy1, TestApp1, usage1)
+		t.Fatalf("unable to increase tracked resource: queuepath %s, app %s, res %v", path1, TestApp1, usage1)
 	}
 
 	groupTracker.setLimits(hierarchy1, resources.Multiply(usage1, 5), 5)
 	groupTracker.setLimits(hierarchy5, resources.Multiply(usage1, 10), 10)
 
-	result = groupTracker.increaseTrackedResource(hierarchy1, TestApp2, usage1, user.User)
+	result = groupTracker.increaseTrackedResource(path1, TestApp2, usage1, user.User)
 	if !result {
-		t.Fatalf("unable to increase tracked resource: queuepath %+q, app %s, res %v", hierarchy1, TestApp2, usage1)
+		t.Fatalf("unable to increase tracked resource: queuepath %s, app %s, res %v", path1, TestApp2, usage1)
 	}
-	result = groupTracker.increaseTrackedResource(hierarchy1, TestApp3, usage1, user.User)
+	result = groupTracker.increaseTrackedResource(path1, TestApp3, usage1, user.User)
 	if !result {
-		t.Fatalf("unable to increase tracked resource: queuepath %+q, app %s, res %v", hierarchy1, TestApp3, usage1)
+		t.Fatalf("unable to increase tracked resource: queuepath %s, app %s, res %v", path1, TestApp3, usage1)
 	}
 	groupTracker.setLimits(hierarchy1, usage1, 1)
 	groupTracker.setLimits(hierarchy5, usage1, 1)
 }
 
+func TestGroupTrackingEvents(t *testing.T) {
+	GetUserManager()
+	user := security.UserGroup{User: "test", Groups: []string{"test"}}
+	eventSystem := mock.NewEventSystem()
+	groupTracker := newGroupTracker(user.User, newUGMEvents(eventSystem))
+	usage := resources.NewResourceFromMap(map[string]resources.Quantity{"cpu": 1})
+
+	groupTracker.increaseTrackedResource(path1, TestApp1, usage, user.User)
+	assert.Equal(t, 1, len(eventSystem.Events))
+	assert.Equal(t, si.EventRecord_UG_GROUP_RESOURCE, eventSystem.Events[0].EventChangeDetail)
+
+	eventSystem.Reset()
+	groupTracker.decreaseTrackedResource(path1, TestApp1, usage, true)
+	assert.Equal(t, 1, len(eventSystem.Events))
+	assert.Equal(t, si.EventRecord_UG_GROUP_RESOURCE, eventSystem.Events[0].EventChangeDetail)
+}
```

Comment:
These asserts(Just 2 lines !) can be added to the existing test cases itself rather than introducing a new method.

Reply:
Done